app/models/period.rb

```diff
@@ -53,12 +53,16 @@ def to_quarter_period
   # Returns a range of dates that correspond to the 'control range' for this period.
   # For example, for a month period of July 1st 2020, this will return the range of April 30th..July 31st.
   def blood_pressure_control_range
-    three_months_ago = end_date.advance(months: -3)
+    three_months_ago = end_date.advance(months: -3).end_of_month
     (three_months_ago..end_date)
   end
 
   alias_method :bp_control_range, :blood_pressure_control_range
 
+  def bp_control_range_registration_date
```

Comment:
Is this the same as `bp_control_range_start_date` but different by one day? I'm wondering if the change on line 56 is all we need we keep using `bp_control_range_start_date`. These two are _very_ similar, I might be missing something.

Reply:
That's right - `bp_control_range_registration_date` is just one day before `bp_control_range_start_date`. I'm creating a new method to make it clear that `bp_control_range` has 3 dates we care about:
* The registration end date
* The start of the BP control range
* The end of the BP control range

How would you implement this?

Reply:
Ah I see. That makes sense now. Maybe `bp_control_registrations_until_date` is a better name since this date is not part of a range really. This is good though 👍🏼.

Reply:
Thanks @prabhanshuguptagit I'll update the name and then merge :)